packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -104,7 +104,12 @@ const MaybeNestedStack = ({
   if (isHeaderInModal) {
     return (
       <ScreenStack style={styles.container}>
-        <Screen enabled style={StyleSheet.absoluteFill}>
+        <Screen
+          enabled
+          isNativeStack
+          hasLargeHeader={options.headerLargeTitle ?? false}
```

Comment:
One comment about `isNativeStack` and `hasLargeHeader` props:
- `isNativeStack` is used internally by react-native-screens mainly for transition progress. It prevents non native-stack navigators from sending transition progress from their Screen components. It was already set on native-stack v5, and it should be also set there.
- hasLargeHeader is a prop that is used by our `ReanimatedNativeStackScreen` component and is used to determine if the screen has large header (since this information cannot be obtained from the sole screen).

Reply:
> isNativeStack is used internally by react-native-screens mainly for transition progress. It prevents non native-stack navigators from sending transition progress from their Screen components. It was already set on native-stack v5, and it should be also set there.

Couldn't it automatically be detected (e.g. with context) when `Screen` is used inside `ScreenStack`?

Reply:
To be fair I wouldn't put a special context just to detect if screen is inside the ScreenStack. What do you mean by detecting the screen with `React.Children`? 🤔

Reply:
Something like `React.Children.map(children, (child) => React.cloneElement(child, { isNativeStack: true })` in `ScreenStack`.

> To be fair I wouldn't put a special context just to detect if screen is inside the ScreenStack

IMO if it's already possible to know if `isNativeStack` needs to be `true`, the responsibility shouldn't fall on the consumer. Otherwise it's to make a mistake and introduce a bug due to refactors later for something that could be handled automatically.